cmd/client/resources/object.go

```diff
@@ -90,6 +91,68 @@ func GetAllObject(cmd *cobra.Command) error {
 	return nil
 }
 
+// EditObject edit an object.
+func EditObject(cmd *cobra.Command, args *general.ArgInfo, kind string) error {
+	getErr := func(err error) error {
+		return general.ErrorMsg(general.EditCmd, err, kind, args.Name)
+	}
+
+	// get old yaml and save it to a temp file
+	var oldYaml string
+	var err error
+	if oldYaml, err = getObjectYaml(args.Name); err != nil {
+		return getErr(err)
+	}
+	filePath := getResourceTempFilePath(kind, args.Name)
```

Comment:
should move this line into `editResource`, there are other cases.

Reply:
`filePath` and remove it are out of function `editResource` purposely. It follows the design of `kubectl edit`. The reason of this design is when edit failed, we want user to find the file and change it manually (for example, some typo, and resource is not updated, or they change the name or kind of resource). So the file is removed only if edit is succeeded and resource is updated as expect.